precompiles/Modexp.yul

```diff
@@ -28,6 +28,100 @@ object "ModExp" {
+            /// @dev Reference: https://github.com/lambdaclass/lambdaworks/blob/main/math/src/unsigned_integer/element.rs#L785
+            /// @param leftLimb The left side of the difference (i.e. the a in a - b).
+            /// @param rightLimb The right side of the difference (i.e. the b in a - b).
+            /// @return substractionResult i.e. the c in c = a - b.
+            /// @return returnBorrow If there was any borrow on the substraction, is returned as 1.
+            function subLimbsWithBorrow(leftLimb, rightLimb, limbBorrow) -> substractionResult, returnBorrow {
+                let rightPlusBorrow := add(rightLimb, limbBorrow)
+                substractionResult := sub(leftLimb, rightPlusBorrow)
+                if gt(substractionResult, leftLimb) {
+                 returnBorrow := 1
+                }
+            }
+            /// @notice Computes the BigUint substraction between the number stored
+            /// in lshPointer and rhsPointer.
+            /// @dev Reference: https://github.com/lambdaclass/lambdaworks/blob/main/math/src/unsigned_integer/element.rs#L795
+            /// @param lhsPointer The start of the left hand side substraction Big Number.
+            /// @param rhsPointer The start of the right hand side substraction Big Number.
+            /// @return numberOfLimbs The number of limbs of both numbers.
+            /// @return resultPointer Where the result will be stored.
+            function bigUintSubstractionWithBorrow(lhsPointer, rhsPointer, numberOfLimbs, resultPointer) -> resultPointer, borrow {
+                let leftIthLimbValue
+                let rightIthLimbValue
+                let ithLimbBorrowResult 
+                let ithLimbSubstractionResult 
+                let borrow := 0
+                let limbOffset := 0
+                for {let i := numberOfLimbs} gt(i, 0) {i := sub(i, 1)} {
+                    limbOffset := mul(sub(i,1), 32)
+                    leftIthLimbValue := getLimbValueAtOffset(lhsPointer, limbOffset)
+                    rightIthLimbValue := getLimbValueAtOffset(rhsPointer, limbOffset)
+                    ithLimbSubstractionResult, borrow :=
+                                               subLimbsWithBorrow(leftIthLimbValue, rightIthLimbValue, borrow)
+                    storeLimbValueAtOffset(resultPointer, limbOffset, ithLimbSubstractionResult)
+
+                }
+            }
+
+
+
+
+            // Test4: Big number 4 limb substraction
+            // First Number:
+            //  - First Limb: 0xFFFF
+            //  - Second Limb: 0xAAAA
+            //  - Third Limb: 0xFFFF
+            //  - Fourth Limb: 0xAAAA
+            // Second Number:
+            //  - First Limb: 0xAAAA
+            //  - Second Limb: 0xFFFF
+            //  - Third Limb: 0xAAAA
+            //  - Fourth Limb: 0xFFF
+            //  
+            // Should print:
+            // 0x5554 (first limb)
+            // 0xffffffffffffaaab (second limb)
```

Comment:
I wasn't getting the same value as the comments suggested, so I made a python script to test it out:

```python
def bigint_from_limbs(a, b, c, d):
    x = a
    x = (x << 256) | b
    x = (x << 256) | c
    x = (x << 256) | d
    return x

def limbs_from_bigint(bi):
    MASK = (1 << 256) - 1
    d = (bi >> (256 * 0)) & MASK
    c = (bi >> (256 * 1)) & MASK
    b = (bi >> (256 * 2)) & MASK
    a = (bi >> (256 * 3)) & MASK
    return (a, b, c, d)

def print_limbs(title, limbs):
    (a, b, c, d) = limbs
    print(f"{title}: [{hex(a)}, {hex(b)}, {hex(c)}, {hex(d)}]")

lhs_limbs = (0xFFFF, 0xAAAA, 0xFFFF, 0xAAAA)
rhs_limbs = (0xAAAA, 0xFFFF, 0xAAAA, 0xFFF)
print_limbs("lhs_limbs", lhs_limbs)
print_limbs("rhs_limbs", rhs_limbs)

lhs = bigint_from_limbs(*lhs_limbs)
rhs = bigint_from_limbs(*rhs_limbs)
assert lhs_limbs == limbs_from_bigint(lhs)
assert rhs_limbs == limbs_from_bigint(rhs)
print("lhs: ", lhs)
print("rhs: ", rhs)

assert lhs > rhs

res = lhs - rhs
print("res: ", res)
res_limbs = limbs_from_bigint(res)
print_limbs("res_limbs", res_limbs)
```

Which gave me the following result:
```
lhs_limbs: [0xffff, 0xaaaa, 0xffff, 0xaaaa]
rhs_limbs: [0xaaaa, 0xffff, 0xaaaa, 0xfff]
lhs:  101744273178926960064988370776390104958760025131742712410997775312037249184302295032822575714652627321637385044653794456059156179857737103542777570440297726813614998909130177654716915530113012821233856222054627051083490363276862118996650
rhs:  67829515452617973376658913850926736639173350087828474940665183541358166122868684844488766469824485205385066374283620948616688854627385354361415216852260078931923727512825878445441091596645481527124071237319796322089793313955471257767935
res:  33914757726308986688329456925463368319586675043914237470332591770679083061433610188333809244828142116252318670370173507442467325230351749181362353588037647881691271396304299209275823933467531294109784984734830728993697049321390861228715
res_limbs: [0x5554, 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffaaab, 0x5555, 0x9aab]
```